### PR TITLE
feat(CX-2457): added correct events to the SWA banner on MC Artwork page

### PR DESCRIPTION
**Pull request by dariakoko**
The type of this PR is: **Enhancement**

This PR resolves [CX-2457]

### Description

Added correct events to the SWA banner on MC Artwork page

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [ ] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Added correct events to the SWA banner on MC Artwork page -daria

</details>

[CX-2457]: https://artsyproduct.atlassian.net/browse/CX-2457?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/MyCollection/Screens/Artwork/Components/MyCollectionWhySell.tsx [hunk 1]
```diff
@@ -24,6 +26,12 @@ export const MyCollectionWhySell: React.FC<MyCollectionWhySellProps> = (props) =
   if (isInProgress || isSold) {
     return null
   }
+  const setContextModule =
+    contextModule === "insights"
+      ? ContextModule.myCollectionArtworkInsights
+      : contextModule === "about"
+      ? ContextModule.myCollectionArtworkAbout
+      : ContextModule.sellFooter
```
Comment (MounirDhahri) [inline src/app/Scenes/MyCollection/Screens/Artwork/Components/MyCollectionWhySell.tsx:34]: **suggestion:** although there is nothing with this I tend to agree with this [article](https://medium.com/@benlmsc/stop-using-nested-ternary-operators-heres-why-53e7e078e65a) that recommends to use if statements or switch statements instead. I will leave it to you and @lordkiz to decide

**Review by MounirDhahri — APPROVED** [review]
LGTM 🙌 Thanks Daria 👏 I left one opinionated comment, I look forward to hear what you and Kizito think about it

File: src/app/Scenes/MyCollection/Screens/Artwork/Components/MyCollectionWhySell.tests.tsx [hunk 2]
```diff
@@ -194,5 +190,134 @@ describe("MyCollectionWhySell", () => {
...
+
+      expect(trackEvent).toHaveBeenCalledWith({
+        action: "tappedLearnMore",
+        context_module: "myCollectionArtworkInsights",
+        context_screen_owner_type: "myCollectionArtwork",
+        context_screen_owner_id: "someInternalId",
+        context_screen_owner_slug: "someSlug",
+        subject: "Learn More",
+      })
+    })
```
Comment (lordkiz) [inline src/app/Scenes/MyCollection/Screens/Artwork/Components/MyCollectionWhySell.tests.tsx:321]: Non-Blocking: Looks like these tests can be refactored into:
```
describe("P1", () => { 
 // P1 related tests
})
describe("Non P1", () => { 
 // Non P1 related tests
})
```

File: src/app/Scenes/MyCollection/Screens/Artwork/Components/MyCollectionWhySell.tsx (hunk 1 above)
Comment (lordkiz) [reply src/app/Scenes/MyCollection/Screens/Artwork/Components/MyCollectionWhySell.tsx:34]: True.
Good old if/else might read better and shorter:
```
let setContextModule = ContextModule.sellFooter
if (contextModule === "insights") {
  setContextModule = ContextModule.myCollectionArtworkInsights
} else if (contextModule === "about") {
  setContextModule = ContextModule.myCollectionArtworkAbout
}
```

**Comment by lordkiz** [comment]
@dariakoko left a few comments. Great work! Let me know when it is ready

File: src/app/Scenes/MyCollection/Screens/Artwork/Components/MyCollectionWhySell.tests.tsx (hunk 2 above)
Comment (lordkiz) [reply src/app/Scenes/MyCollection/Screens/Artwork/Components/MyCollectionWhySell.tests.tsx:321]: And the `it()` descriptions can be better. For eg `it("insights tab, NOT P1")` does not describe what insights tab is supposed to do or be.

**Review by MounirDhahri — APPROVED** [review]
Thank you for the changes Daria 🙌

**Comment by lordkiz** [comment]
#squashongreen

**Pull request merged**